From 2c230795802b1f456c7c8604a465ed5a2391140e Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Thu, 12 Jun 2014 19:41:39 +0300
Subject: [PATCH 0036/1240] a8k: restructure the configuration file

	- remove the unnecessary configs, and move the shared one to
	  mvebu-common.h
	- add base address of the UART
	- update the soc_tclk_get() to return the defined clock frequency
	- re-define the baudrate for a8k

Change-Id: Ie45ae887120bdb14ef31e9b4e90c2128ef43fa5f
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/8513
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
---
 arch/arm/cpu/armv8/armada8k/clock.c            |   2 +-
 arch/arm/include/asm/arch-armada8k/regs-base.h |   4 +
 include/configs/armada38x.h                    |   5 -
 include/configs/armada8k.h                     | 192 ++++---------------------
 include/configs/mvebu-common.h                 |  18 ++-
 5 files changed, 50 insertions(+), 171 deletions(-)

diff --git a/arch/arm/cpu/armv8/armada8k/clock.c b/arch/arm/cpu/armv8/armada8k/clock.c
index fc83ed6..b2c5dce 100644
--- a/arch/arm/cpu/armv8/armada8k/clock.c
+++ b/arch/arm/cpu/armv8/armada8k/clock.c
@@ -22,7 +22,7 @@
 
 u32 soc_tclk_get(void)
 {
-	return 200000000;
+	return COUNTER_FREQUENCY;
 }
 
 u32 soc_cpu_clk_get(void)
diff --git a/arch/arm/include/asm/arch-armada8k/regs-base.h b/arch/arm/include/asm/arch-armada8k/regs-base.h
index 4d326f6..a7d4dae 100644
--- a/arch/arm/include/asm/arch-armada8k/regs-base.h
+++ b/arch/arm/include/asm/arch-armada8k/regs-base.h
@@ -27,11 +27,15 @@
 /* List of register base for all units */
 #define MVEBU_ADEC_AP_BASE (MVEBU_REGS_BASE + 0x0)
 
+#define MVEBU_UART_BASE(x)	(MVEBU_REGS_BASE + 0x512000 + (0x100 * x))
 #define MVEBU_MISC_REGS_BASE	(MVEBU_REGS_BASE + 0x18200)
 #define MVEBU_DEVICE_ID_REG	(MVEBU_MISC_REGS_BASE + 0x38)
 #define MVEBU_DEVICE_REV_REG	(MVEBU_MISC_REGS_BASE + 0x3C)
 #define MVEBU_RESET_MASK_REG	(MVEBU_MISC_REGS_BASE + 0x60)
 #define MVEBU_SOFT_RESET_REG	(MVEBU_MISC_REGS_BASE + 0x64)
 
+#define MVEBU_GICD_BASE		(0x2C001000)
+#define MVEBU_GICC_BASE		(0x2C002000)
+
 
 #endif	/* _ARMADA8K_H_ */
diff --git a/include/configs/armada38x.h b/include/configs/armada38x.h
index 1baeb7a..678f234 100644
--- a/include/configs/armada38x.h
+++ b/include/configs/armada38x.h
@@ -86,7 +86,6 @@
 #define CONFIG_OF_BOARD_SETUP	1
 */
 
-
 /* Clock */
 #ifndef __ASSEMBLY__
 	#define CONFIG_SYS_HZ                   1000
@@ -97,12 +96,8 @@
 /* DRAM detection stuff */
 #define MV_DRAM_AUTO_SIZE
 #define CONFIG_SYS_DRAM_BANKS           1
-#define CONFIG_NR_DRAM_BANKS            1
-#define CONFIG_SYS_MEMTEST_START        0x00400000
-#define CONFIG_SYS_MEMTEST_END          0x007fffff
 
 /* Include AFTER since it is affected by defines above */
 #include "mvebu-common.h"
 
-
 #endif /* __CONFIG_H */
diff --git a/include/configs/armada8k.h b/include/configs/armada8k.h
index 9d9db93..4fbdee5 100644
--- a/include/configs/armada8k.h
+++ b/include/configs/armada8k.h
@@ -20,167 +20,43 @@
 #ifndef __ARMADA_8K_H
 #define __ARMADA_8K_H
 
-#include "mvebu-common.h"
+/* Version */
+#define CONFIG_IDENT_STRING             " armada8021-pxp"
+#define CONFIG_BOOTP_VCI_STRING         "U-boot.armv8.armada8021-pxp"
 
-#undef CONFIG_SYS_INIT_SP_ADDR
-#undef CONFIG_SYS_MALLOC_LEN
-#undef CONFIG_SYS_LOAD_ADDR
-#undef CONFIG_SYS_SDRAM_BASE
-#undef CONFIG_ENV_IS_NOWHERE
+/* Enable IO drivers */
+#define MV_INCLUDE_UART
+/*
+#define MV_INCLUDE_SDIO
+#define MV_INCLUDE_INTEG_SATA
+#define MV_INCLUDE_CLK_PWR_CNTRL
+
+#define MV_INCLUDE_RCVR
+#define MV_INCLUDE_NAND
+#define MV_INCLUDE_SPI
+#define MV_INCLUDE_XOR
+#define MV_INCLUDE_SATA
+#define MV_INCLUDE_USB
+#define CONFIG_MV_ETH_NETA
+#define MV_INCLUDE_GPIO
+#define MV_INCLUDE_I2C
+#define MV_INCLUDE_RTC
+#define MV_INCLUDE_PEX
+*/
 
 #define CONFIG_AURORA_2_TIMER
-
-#define DEBUG
-
 #define CONFIG_REMAKE_ELF
-
-/*#define CONFIG_ARMV8_SWITCH_TO_EL1*/
-
-/*#define CONFIG_SYS_GENERIC_BOARD*/
-
-/* Has ADEC address decode unit */
-#define CONFIG_ADEC
-
-#define CONFIG_SYS_NO_FLASH
-
-#define CONFIG_SUPPORT_RAW_INITRD
-
-/* Cache Definitions */
-#define CONFIG_SYS_DCACHE_OFF
 #define CONFIG_SYS_ICACHE_OFF
-
-#define CONFIG_IDENT_STRING		" armada8021-pxp"
-#define CONFIG_BOOTP_VCI_STRING		"U-boot.armv8.armada8021-pxp"
-
-/* Link Definitions */
-#define CONFIG_SYS_TEXT_BASE		0x80000000
-#define CONFIG_SYS_INIT_SP_ADDR         (CONFIG_SYS_SDRAM_BASE + 0x7fff0)
-
-/* Flat Device Tree Definitions */
-#define CONFIG_OF_LIBFDT
-
-#define CONFIG_DEFAULT_DEVICE_TREE	vexpress64
-
-/* SMP Spin Table Definitions */
-#define CPU_RELEASE_ADDR		(CONFIG_SYS_SDRAM_BASE + 0x7fff0)
-
-/* CS register bases for the original memory map. */
-#define V2M_PA_CS0			0x00000000
-#define V2M_PA_CS1			0x14000000
-#define V2M_PA_CS2			0x18000000
-#define V2M_PA_CS3			0x1c000000
-#define V2M_PA_CS4			0x0c000000
-#define V2M_PA_CS5			0x10000000
-
-#define V2M_PERIPH_OFFSET(x)		(x << 16)
-#define V2M_SYSREGS			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(1))
-#define V2M_SYSCTL			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(2))
-#define V2M_SERIAL_BUS_PCI		(V2M_PA_CS3 + V2M_PERIPH_OFFSET(3))
-
-#define V2M_BASE			0x80000000
-
-/*
- * Physical addresses, offset from V2M_PA_CS0-3
- */
-#define V2M_NOR0			(V2M_PA_CS0)
-#define V2M_NOR1			(V2M_PA_CS4)
-#define V2M_SRAM			(V2M_PA_CS1)
-
-/* Common peripherals relative to CS7. */
-#define V2M_AACI			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(4))
-#define V2M_MMCI			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(5))
-#define V2M_KMI0			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(6))
-#define V2M_KMI1			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(7))
-
-#define V2M_UART0			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(9))
-#define V2M_UART1			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(10))
-#define V2M_UART2			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(11))
-#define V2M_UART3			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(12))
-
-#define V2M_WDT				(V2M_PA_CS3 + V2M_PERIPH_OFFSET(15))
-
-#define V2M_TIMER01			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(17))
-#define V2M_TIMER23			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(18))
-
-#define V2M_SERIAL_BUS_DVI		(V2M_PA_CS3 + V2M_PERIPH_OFFSET(22))
-#define V2M_RTC				(V2M_PA_CS3 + V2M_PERIPH_OFFSET(23))
-
-#define V2M_CF				(V2M_PA_CS3 + V2M_PERIPH_OFFSET(26))
-
-#define V2M_CLCD			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(31))
-
-/* System register offsets. */
-#define V2M_SYS_CFGDATA			(V2M_SYSREGS + 0x0a0)
-#define V2M_SYS_CFGCTRL			(V2M_SYSREGS + 0x0a4)
-#define V2M_SYS_CFGSTAT			(V2M_SYSREGS + 0x0a8)
-
 /* Generic Timer Definitions */
-#define COUNTER_FREQUENCY		(0x1800000)	/* 24MHz */
+#define COUNTER_FREQUENCY               (1008000)     /* 24MHz */
 
-/* Generic Interrupt Controller Definitions */
-#define GICD_BASE			(0x2C001000)
-#define GICC_BASE			(0x2C002000)
-
-#define CONFIG_SYS_MEMTEST_START	V2M_BASE
-#define CONFIG_SYS_MEMTEST_END		(V2M_BASE + 0x80000000)
-
-/* Size of malloc() pool */
-#define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + 128 * 1024)
-
-/* SMSC91C111 Ethernet Configuration */
-/*#define CONFIG_SMC91111			1
-#define CONFIG_SMC91111_BASE		(0x01A000000)*/
-
-/* PL011 Serial Configuration */
-#define CONFIG_PL011_SERIAL
-#define CONFIG_PL011_CLOCK		24000000
-#define CONFIG_PL01x_PORTS		{(void *)CONFIG_SYS_SERIAL0, \
-					 (void *)CONFIG_SYS_SERIAL1}
-#define CONFIG_CONS_INDEX		0
-
-#define CONFIG_BAUDRATE			38400
-#define CONFIG_SYS_BAUDRATE_TABLE	{ 9600, 19200, 38400, 57600, 115200 }
-#define CONFIG_SYS_SERIAL0		V2M_UART0
-#define CONFIG_SYS_SERIAL1		V2M_UART1
-
-/* Command line configuration */
-#define CONFIG_MENU
-/*#define CONFIG_MENU_SHOW*/
-#define CONFIG_CMD_CACHE
-#define CONFIG_CMD_BDI
-#define CONFIG_CMD_DHCP
-#define CONFIG_CMD_PXE
-#define CONFIG_CMD_ENV
-#define CONFIG_CMD_FLASH
-#define CONFIG_CMD_IMI
-#define CONFIG_CMD_MEMORY
-#define CONFIG_CMD_MII
-#define CONFIG_CMD_NET
-#define CONFIG_CMD_PING
-#define CONFIG_CMD_SAVEENV
-#define CONFIG_CMD_RUN
-#define CONFIG_CMD_BOOTD
-#define CONFIG_CMD_ECHO
-#define CONFIG_CMD_SOURCE
-#define CONFIG_CMD_FAT
-#define CONFIG_DOS_PARTITION
-
-/* BOOTP options */
-#define CONFIG_BOOTP_BOOTFILESIZE
-#define CONFIG_BOOTP_BOOTPATH
-#define CONFIG_BOOTP_GATEWAY
-#define CONFIG_BOOTP_HOSTNAME
-#define CONFIG_BOOTP_PXE
-#define CONFIG_BOOTP_PXE_CLIENTARCH	0x100
+#include "mvebu-common.h"
 
-/* Miscellaneous configurable options */
-#define CONFIG_SYS_LOAD_ADDR		(V2M_BASE + 0x10000000)
+#undef CONFIG_BOOTDELAY
+#define CONFIG_BOOTDELAY                -1
 
-/* Physical Memory Map */
-#define CONFIG_NR_DRAM_BANKS		1
-#define PHYS_SDRAM_1			(V2M_BASE)	/* SDRAM Bank #1 */
-#define PHYS_SDRAM_1_SIZE		0x80000000	/* 2048 MB */
-#define CONFIG_SYS_SDRAM_BASE		PHYS_SDRAM_1
+#undef CONFIG_BAUDRATE
+#define CONFIG_BAUDRATE                 63000
 
 /* Initial environment variables */
 #define CONFIG_EXTRA_ENV_SETTINGS	\
@@ -194,16 +70,4 @@
 #define CONFIG_BOOTCOMMAND		"bootm $kernel_addr " \
 					"$initrd_addr:$initrd_size $fdt_addr"
 
-/* Do not preserve environment */
-#define CONFIG_ENV_IS_NOWHERE		1
-
-/* Monitor Command Prompt */
-#define CONFIG_SYS_PBSIZE		(CONFIG_SYS_CBSIZE + \
-					sizeof(CONFIG_SYS_PROMPT) + 16)
-#define CONFIG_SYS_HUSH_PARSER
-#define CONFIG_SYS_PROMPT_HUSH_PS2	"> "
-#define CONFIG_SYS_BARGSIZE		CONFIG_SYS_CBSIZE
-#define CONFIG_SYS_LONGHELP
-#define CONFIG_CMDLINE_EDITING		1
-
 #endif /* __ARMADA_8K_H */
diff --git a/include/configs/mvebu-common.h b/include/configs/mvebu-common.h
index 5fb2786..a37983f 100644
--- a/include/configs/mvebu-common.h
+++ b/include/configs/mvebu-common.h
@@ -44,7 +44,9 @@
 #define CONFIG_SYS_RESET_ADDRESS        0xffff0000
 #define CONFIG_SYS_MALLOC_BASE          (CONFIG_SYS_TEXT_BASE + (3 << 20))  /* TEXT_BASE + 3M */
 #define CONFIG_SYS_MALLOC_LEN           (5 << 20)    /* Reserve 5MB for malloc*/
-#define CONFIG_SYS_INIT_SP_ADDR         0x00FF0000   /* End of 16M scrubbed by training in bootrom */
+#define CONFIG_NR_DRAM_BANKS		(4)
+/* maybe need to set back to 0x7fff0 */
+#define CONFIG_SYS_INIT_SP_ADDR         (CONFIG_SYS_TEXT_BASE + 0xFF0000)   /* End of 16M scrubbed by training in bootrom */
 #define CONFIG_SYS_GBL_DATA_SIZE        128          /* Size in bytes reserved for init data */
 #define CONFIG_SYS_MAXARGS		32      /* Max number of command argg */
 #define CONFIG_UBOOT_SIZE		0x100000
@@ -54,6 +56,10 @@
 #define CONFIG_UBOOT_MAX_MEM_SIZE	(3ll << 30)	/* Limit u-boot to 3GB */
 #define MVEBU_IO_RESERVE_BASE		0xC0000000ll
 
+/* memtest definition */
+#define CONFIG_SYS_MEMTEST_START        (CONFIG_SYS_SDRAM_BASE)
+#define CONFIG_SYS_MEMTEST_END          (CONFIG_SYS_SDRAM_BASE + 0x10000000)
+
 /* Board init functions */
 #define CONFIG_BOARD_EARLY_INIT_F	/* pre relloc board init */
 #define CONFIG_MISC_INIT_R              /* post relloc board init */
@@ -63,6 +69,7 @@
 #undef  CONFIG_WATCHDOG                 /* watchdog disabled */
 
 /* Cache */
+#define CONFIG_SYS_DCACHE_OFF
 #define CONFIG_SYS_CACHELINE_SIZE       32
 
 /*
@@ -128,6 +135,15 @@
 	#define CONFIG_SYS_NO_FLASH
 #endif
 
+/* Generic Interrupt Controller Definitions */
+#define GICD_BASE                       MVEBU_GICD_BASE
+#define GICC_BASE                       MVEBU_GICC_BASE
+
+/* Flat Device Tree Definitions */
+#define CONFIG_OF_LIBFDT
+
+/* SMP Spin Table Definitions */
+#define CPU_RELEASE_ADDR               (CONFIG_SYS_SDRAM_BASE + 0x7fff0)
 
 /*********  IO Definitions **********/
 
-- 
1.9.1

